### ISSUE TITLE
Support continuations for category members

### ISSUE THREAD

**Issue by 67yuhjnm** [siznax/wptools#99]
Hello, there are two questions which I need to ask for your help.

a. I want to get the all category members, but the limit is 500. How should I do this?

```python
>>> cat = wptools.category('Category:BAFTA winners (people)')
>>> cat.get_members()
en.wikipedia.org (categorymembers) Category:BAFTA winners (people)
Category:BAFTA winners (people) (en) data
{
  members: <list(500)> {'pageid': 22167530, 'ns': 0, 'title': 'All...
  requests: <list(1)> category
}
```

b. How can I get the "Subcategories" in page, such as "https://en.wikipedia.org/wiki/Category:BAFTA_winners_(people)"

Thanks in advance

**Comment by 67yuhjnm** [comment]
The second question, subcategories list is behind the categories list in the results using the method in first question.

There is only the first question.

**Comment by siznax** [comment]
Hi, @67yuhjnm. Thanks for trying wptools, and for opening this issue. You are right. We should support continuations for category members, and also be able to get subcategories. I'll work on continuations in this issue, and open a new issue for getting subcategories.

**Comment by 67yuhjnm** [comment]
@siznax Thank you very much. 
The limit of category members is 500, is there a method to get the all category members? 
The category members of "Category:BAFTA winners(people)" are 758.

**Comment by siznax** [comment]
@67yuhjnm that (500) is the limit of category members we can get _in one request_ (as described in [API:Categorymembers](https://www.mediawiki.org/wiki/API:Categorymembers)). So, if you look at `cat.query('category')` and follow [the link](https://en.wikipedia.org/w/api.php?action=query&formatversion=2&list=categorymembers&cmlimit=500&cmtitle=Category%3ABAFTA%20winners%20%28people%29):

```python
>>> cat.query('category')
'https://en.wikipedia.org/w/api.php?action=query&formatversion=2&list=categorymembers&cmlimit=500&cmtitle=Category%3ABAFTA%20winners%20%28people%29'
```

you'll see `cmcontinue` in the response:

```
    "continue": {
        "cmcontinue": "page|412745372f4907044b352f2f41273d03424b352f2f412704412745372f490121018f7f8f7f8f808f09|42525291"
```

So, we just need to make more requests until `cmcontinue` is empty. The next request would look like this:

https://en.wikipedia.org/w/api.php?action=query&formatversion=2&list=categorymembers&cmlimit=500&cmtitle=Category%3ABAFTA%20winners%20%28people%29&cmcontinue=page|412745372f4907044b352f2f41273d03424b352f2f412704412745372f490121018f7f8f7f8f808f09|42525291

And I see the subcategories (`ns: 14`) at the end of the response too! So, this shouldn't be difficult.

**Comment by 67yuhjnm** [comment]
@siznax Thank you very much.

**Comment by siznax** [comment]
I've added support for category continuations and put subcategories in a new `subcategories` data member with the commits referenced above:

```python
en.wikipedia.org (categorymembers) Category:BAFTA winners (people)
Category:BAFTA winners (people) (en) data
{
  cmcontinue: <str(96)> page|412745372f4907044b352f2f41273d03424b3...
  members: <list(500)> {u'ns': 0, u'pageid': 22167530, u'title': u...
  requests: <list(1)> category
}
en.wikipedia.org (categorymembers) Category:BAFTA winners (people)
Category:BAFTA winners (people) (en) data
{
  members: <list(768)> {u'ns': 0, u'pageid': 22167530, u'title': u...
  requests: <list(2)> category, category
  subcategories: <list(9)> {u'ns': 14, u'pageid': 23488553, u'titl...
}
```

So, this also fixes #100.

**Comment by 67yuhjnm** [comment]
Thank you very much.
